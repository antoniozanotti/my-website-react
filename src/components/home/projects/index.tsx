import {useTranslations} from 'next-intl';
import Section from "@/components/ui/section";
import WorksList from "@/components/common/work-list";

export default function Index({ projects }: { projects: any }) {
  const t = useTranslations('Projects');
  return (
    <Section title={t('title')}>
      <WorksList
        works={projects}
        labelForMore={t('labelForMore')}
      />
    </Section>
  );
}
