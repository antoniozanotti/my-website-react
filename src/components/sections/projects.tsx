import Section from "@/components/ui/section";
import Heading from "@/components/ui/heading";
import Grid from "@/components/ui/grid";
import Cols from "@/components/ui/cols";
import List from "@/components/ui/list";
import { styles } from "@/components/utils/styles";
import TopazBrandSvg from "@images/topaz_brand";
import VueBrandSvg from "@images/topaz_brand";
import ReactBrandSvg from "@images/topaz_brand";

export default function Projects() {
  return (
    <Section title="# Projects">
      <article>
        <Heading level="h3" title="## Topaz" />
        <Grid>
          <Cols className="mt-4">
            <div className="grow-0">
              <TopazBrandSvg />
            </div>
            <div className="grow">
              <p>
                UI component library for Vue or React projects. Features:
                components, color customization, dark and light theme,
                responsive. Topaz is a MIT licensed project. This website was
                built with Topaz for React.
              </p>
            </div>
          </Cols>
          <div className="mt-4 pl-[46px] sm:pl-[0px]">
            <p>Inspired In:</p>
            <List className="mt-4">
              <li>
                <a
                  href="https://spectrum.adobe.com/"
                  title=""
                  target="_blank"
                  className={styles.aHover}
                >
                  Spectrum
                </a>
                , Adobe’s design system
              </li>
              <li>
                <a
                  href="https://github.com/Decathlon/vitamin-web"
                  title=""
                  target="_blank"
                  className={styles.aHover}
                >
                  Vitamin
                </a>
                , Decathlon’s design system
              </li>
              <li>
                <a
                  href="https://github.com/vuetifyjs/vuetify"
                  title=""
                  target="_blank"
                  className={styles.aHover}
                >
                  Vuetify
                </a>
                , Vue Component Framework
              </li>
            </List>
          </div>
        </Grid>
        <Grid>
          <article>
            <Heading level="h4" title="### Topaz for Vue" />
            <Cols className="mt-4">
              <div className="grow-0">
                <VueBrandSvg />
              </div>
              <div>
                <p>
                  Technologies: Vue, UnoCSS, heroicons, TypeScript, Vite, Git,
                  npm, storybook...
                </p>
                <List className="mt-4">
                  <li>
                    <a
                      href="https://topaz-vue.netlify.app/"
                      title=""
                      target="_blank"
                      className={styles.aHover}
                    >
                      Storybook Documentation
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://github.com/antoniozanotti/topaz-vue"
                      title=""
                      target="_blank"
                      className={styles.aHover}
                    >
                      Github Repository
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://www.npmjs.com/package/topaz-vue"
                      title=""
                      target="_blank"
                      className={styles.aHover}
                    >
                      NPM Package
                    </a>
                  </li>
                </List>
              </div>
            </Cols>
          </article>
          <article>
            <Heading level="h4" title="### Topaz for React" />
            <Cols className="mt-4">
              <div className="grow-0">
                <ReactBrandSvg />
              </div>
              <div>
                <p>
                  Technologies: React, tailwindcss, heroicons, TypeScript, tsdx,
                  Git, npm, storybook...
                </p>
                <List className="mt-4">
                  <li>
                    <a
                      href="https://topaz-react.netlify.app/"
                      title=""
                      target="_blank"
                      className={styles.aHover}
                    >
                      Storybook Documentation
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://github.com/antoniozanotti/topaz-react"
                      title=""
                      target="_blank"
                      className={styles.aHover}
                    >
                      Github Repository
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://www.npmjs.com/package/topaz-react"
                      title=""
                      target="_blank"
                      className={styles.aHover}
                    >
                      NPM Package
                    </a>
                  </li>
                </List>
              </div>
            </Cols>
          </article>
        </Grid>
      </article>
      <article>
        <Heading level="h3" title="## MAD Design Group Ecommerces" />
        <Grid className="mt-4">
          <div className="flex flex-col space-y-4">
            <p>
              For 8 years, I participated in the web development team of MAD
              Design Group, working as a Full Stack Magento Developer.
            </p>
            <p>
              In July 2015, when I was hired, I had the challenge of creating a
              centralized and scalable base for all MAD’s ecommerces. At that
              time, each brand and location had its own code, which made
              maintenance and improvement impractical. This project was built in
              gradual development cycles, until all brands and locations were
              incorporated into the same code. Currently the project has 30
              ecommerces, 5 brands in 6 locations. Below are some of them:
            </p>
            <List className="pl-4">
              <li>
                <a
                  href="https://ecosmartfire.com/"
                  title=""
                  target="_blank"
                  className={styles.aHover}
                >
                  EcoSmart Fire (United States)
                </a>
              </li>
              <li>
                <a
                  href="https://blindedesign.ca/"
                  title=""
                  target="_blank"
                  className={styles.aHover}
                >
                  Blinde Design (Canada)
                </a>
              </li>
              <li>
                <a
                  href="https://e-nrg.com.au/"
                  title=""
                  target="_blank"
                  className={styles.aHover}
                >
                  e-NRG Bioethanol (Australia)
                </a>
              </li>
              <li>
                <a
                  href="https://heatscopeheaters.eu/"
                  title=""
                  target="_blank"
                  className={styles.aHover}
                >
                  HEATSCOPE® Heaters (Europe)
                </a>
              </li>
              <li>
                <a
                  href="https://mad-gl.com/"
                  title=""
                  target="_blank"
                  className={styles.aHover}
                >
                  MAD Design Group (Internacional)
                </a>
              </li>
            </List>
            <p>
              The user experience in ecommerces have always been a relevant
              factor, and therefore, numerous modules have been developed to
              meet customization needs.
            </p>
            <p>
              Some internal processes were automated, such as exporting orders
              and leads to Netsuite (ERP).
            </p>
          </div>
          <div className="flex flex-col space-y-4">
            <p>
              In addition to ecommerces for end customers, online stores with
              restricted access for B2B were developed, with specific features
              that I do not list here for ethical reasons.
            </p>
            <p>
              In 2018 and 2019, I participated in the migration from Magento 1
              to Magento 2. Due to the high customization and small team, it
              took 1 year to do so, adding up the planning and execution time of
              the project. As if the complexity of code refactoring and data
              migration wasn’t enough, in the same period, the layout of
              ecommerces was also changed.
            </p>
            <p>
              Other less complex tasks were also developed, such as monitoring,
              preventing and inhibiting attacks. Updating PHP and Magento
              versions, SEO optimization…
            </p>
            <p>
              In 2023, we began a migration project to a Jamstack architecture,
              which use Vue. I developed the initial set-up of the project so
              that everyone could develop collaboratively.
            </p>
          </div>
        </Grid>
      </article>
    </Section>
  );
}
