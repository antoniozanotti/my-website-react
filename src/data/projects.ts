const projects = [
  {
    title: "Topaz UI component library",
    companyName: "Topaz",
    period: "Nov 2023 - Present",
    version: "short",
    achievements: [
      "Develop and maintain this UI component library for React projects using Next.js, TypeScript, Tailwind CSS, Jest and Testing Library for Unit Testing, tsdx for packaging, Storybook for documentation. Go to GitHub, NPM or Documentation.",
      "Maintained the version of this library for Vue projects using Vite, TypeScript, UnoCSS with Tailwind CSS, Vitest and Testing Library for Unit Testing. Go to GitHub, NPM or Documentação.",
      "Fully responsive, dark and light theme, customizable through preset (font size and colors), reusable components (button, heading, icon, input, textarea, label and toogle theme so far).",
    ],
    links: [
      {
        link: "https://github.com/antoniozanotti/topaz-react",
        label: "GitHub (React)",
      },
      {
        link: "https://www.npmjs.com/package/topaz-react",
        label: "npm (React)",
      },
      {
        link: "https://topaz-react.netlify.app/",
        label: "Storybook (React)",
      },
      {
        link: "https://github.com/antoniozanotti/topaz-vue",
        label: "GitHub (Vue)",
      },
      {
        link: "https://www.npmjs.com/package/topaz-vue",
        label: "npm (Vue)",
      },
      {
        link: "https://topaz-vue.netlify.app/",
        label: "Storybook (Vue)",
      },
    ],
    skills: [
      "Front-End Development",
      "React",
      "Vue",
      "Next.js",
      "Vite",
      "TypeScript",
      "JavaScript",
      "Tailwind CSS",
      "CSS",
      "HTML",
      "npm",
      "Storybook",
      "Git",
      "Test Automation",
      "Unit Testing",
      "Jest",
      "Vitest",
    ],
  },
  {
    title: "Migration to Jamstack archicture",
    companyName: "MAD Design Group",
    period: "Jan 2023 - Dec 2023",
    version: "short",
    achievements: [
      "Proposed the transition to the Jamstack architecture, gradually replacing the Magento platform with APIs. This migration will result in significant improvements in application performance, increased productivity in the development environment and will provide complete independence to content managers.",
      "Set up a Vue project for collaborative development, allowing the entire team to build reusable pages and components. I integrated Storybook into the project for documentation purposes and isolated development of components, ensuring consistency in the development process.",
      "Analyzed and proposed improvements to data modeling in Hygraph (GraphQL Headless CMS), aiming to improve data organization to increase efficiency and flexibility on the platform.",
    ],
    skills: [
      "Data Modeling",
      "GraphQL",
      "Front-End Development",
      "TypeScript",
      "JavaScript",
      "Tailwind CSS",
      "CSS",
      "HTML",
      "Vite",
      "Vue",
      "Git",
      "Storybook",
      "Pinia",
    ],
  },
];

export default projects;
