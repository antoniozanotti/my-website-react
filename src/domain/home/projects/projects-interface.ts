import { ArticleType } from "@/types/article-type";

export interface ProjectsInterface {
  projects: ArticleType[];
}
