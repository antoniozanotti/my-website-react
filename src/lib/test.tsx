import React, { ReactElement } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";
import { Theme } from "@/domain/theme/theme/theme";
import { Providers } from "../app/providers";
import messages from "@/messages/en.json";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return (
    <Providers locale="en" messages={messages}>
      <Theme>
        <div data-testid="container">{children}</div>
      </Theme>
    </Providers>
  );
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
  };
});
