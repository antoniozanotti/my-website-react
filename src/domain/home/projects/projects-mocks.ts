import { ProjectsInterface } from "@/domain/home/projects/projects-interface";

export const ProjectsMocks: ProjectsInterface = {
  projects: [
    {
      __typename: "Project",
      title: "Emmanuel's Book Store",
      company: null,
      period: "Apr 2024 - Present",
      visibleInShortVersion: true,
      achievements: {
        __typename: "ProjectAchievements",
        json: {
          nodeType: "document",
          data: {},
          content: [
            {
              nodeType: "paragraph",
              data: {},
              content: [
                {
                  nodeType: "text",
                  value:
                    "React ecommerce for portfolio purposes, my biggest project created from scratch with Next.js to show what I can do. I highlight the search and shopping cart features.",
                  marks: [],
                  data: {},
                },
              ],
            },
          ],
        },
      },
      skillsCollection: {
        __typename: "ProjectSkillsCollection",
        items: [
          { __typename: "Skill", title: "Front-End Development" },
          { __typename: "Skill", title: "Software Architecture" },
          { __typename: "Skill", title: "React" },
          { __typename: "Skill", title: "Next.js" },
          { __typename: "Skill", title: "Node.js" },
          { __typename: "Skill", title: "TypeScript" },
          { __typename: "Skill", title: "JavaScript" },
          { __typename: "Skill", title: "TanStack Query" },
          { __typename: "Skill", title: "Axios" },
          { __typename: "Skill", title: "Test Automation" },
          { __typename: "Skill", title: "Playwright" },
          { __typename: "Skill", title: "Regex" },
          { __typename: "Skill", title: "Data Modeling" },
          { __typename: "Skill", title: "PostgreSQL" },
          { __typename: "Skill", title: "Prisma" },
          { __typename: "Skill", title: "SQL" },
          { __typename: "Skill", title: "HTML" },
          { __typename: "Skill", title: "Tailwind CSS" },
          { __typename: "Skill", title: "CSS" },
          { __typename: "Skill", title: "NextUI" },
          { __typename: "Skill", title: "SEO" },
          { __typename: "Skill", title: "Software Development" },
          { __typename: "Skill", title: "Documentation" },
          { __typename: "Skill", title: "Faker.js" },
          { __typename: "Skill", title: "Git" },
          { __typename: "Skill", title: "npm" },
        ],
      },
    },
  ],
};
