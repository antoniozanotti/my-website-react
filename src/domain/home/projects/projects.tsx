import { ProjectsInterface } from "./projects-interface";
import { useTranslations } from "next-intl";
import { Section } from "@/domain/ui/section/section";
import { ArticleListContainer } from "../article-list/article-list-container";

export function Projects({ projects }: ProjectsInterface) {
  const t = useTranslations("Projects");

  return (
    <Section title={t("title")}>
      <ArticleListContainer articles={projects} labelForMore={t("labelForMore")} />
    </Section>
  );
}
