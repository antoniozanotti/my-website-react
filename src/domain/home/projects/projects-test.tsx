import { test, expect } from '@playwright/experimental-ct-react';
import { Projects } from "./projects";
import { ProjectsMocks } from './projects-mocks';

test.describe("Projects", () => {
  test("should render section", async ({ mount }) => {
    const component = await mount(<Projects projects={ProjectsMocks.projects} />);
    await expect(component.getByText("Projects")).toBeVisible();
  });
});
