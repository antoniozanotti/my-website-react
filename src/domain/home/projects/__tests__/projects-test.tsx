import { screen, render } from "@/lib/test";
import { Projects } from "../projects";
import { ProjectsMocks } from './projects-mocks';

describe("Projects", () => {
  it("should render section", async () => {
    const { container } = render(<Projects projects={ProjectsMocks.projects} />);
    const title = screen.getByRole("heading", {name: "Projects"});
    
    expect(title).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
