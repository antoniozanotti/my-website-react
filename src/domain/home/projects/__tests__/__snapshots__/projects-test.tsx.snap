// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects should render section 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <section>
          <h2
            class="leading-none text-c9 sm:text-c10 lg:text-c11 text-accent-1 dark:text-dark-accent-1 drop-shadow mt-[164px] "
            id=""
          >
            Projects
          </h2>
          <div>
            <div>
              <div
                style="opacity: 0;"
              >
                <h3
                  class="leading-none text-c7 sm:text-c8 lg:text-c10 text-accent-2 dark:text-dark-accent-2 drop-shadow mt-[68px] "
                  id=""
                >
                  Emmanuel's Book Store
                </h3>
                <div
                  class="sm:grid sm:gap-x-8 lg:gap-x-16 sm:grid-cols-3 mt-4 space-y-4"
                >
                  <div
                    class="grid grid-flow-row auto-rows-max"
                  >
                    <span />
                    <span
                      class="font-bold py-4"
                    >
                      Apr 2024 - Present
                    </span>
                    <div
                      class="flex flex-wrap gap-2"
                    >
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Front-End Development
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Software Architecture
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        React
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Next.js
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Node.js
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        TypeScript
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        JavaScript
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        TanStack Query
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Axios
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Test Automation
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Playwright
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Regex
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Data Modeling
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        PostgreSQL
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Prisma
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        SQL
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        HTML
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Tailwind CSS
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        CSS
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        NextUI
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        SEO
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Software Development
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Documentation
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Faker.js
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Git
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        npm
                      </span>
                    </div>
                  </div>
                  <div
                    class="col-span-2 space-y-4"
                  >
                    <div
                      class="space-y-4"
                    >
                      <p
                        style="opacity: 0;"
                      >
                        React ecommerce for portfolio purposes, my biggest project created from scratch with Next.js to show what I can do. I highlight the search and shopping cart features.
                         
                      </p>
                    </div>
                    <ul
                      class="list-disc list-inside pl-4"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
